### PR TITLE
Add TensorRT-Model-Optimizer INT4 AWQ support in onnxruntime tools

### PR DESCRIPTION
**Pull request by anujj**
[TensorRT-Model-Optimizer](https://github.com/NVIDIA/TensorRT-Model-Optimizer) have a implementation for INT4 AWQ. Adding the support in onnxruntime tools to quantized the models with TensorRT-Model-Optimizer

### CODE REVIEW

**Comment by anujj** [comment]
@microsoft-github-policy-service agree company="Nvidia"

**Comment by gedoensmax** [comment]
@tianleiwu or @jywu-msft could you review this or redirect to the right people ?

**Comment by tianleiwu** [comment]
Please fix python format. See https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md#linting

File: onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py [hunk 1]
```diff
@@ -1073,10 +1334,35 @@ def parse_args():
...
+        type=str,
+        required=False,
+        help="Path of the tokenizer dir.",
+    )
+    parser.add_argument(
+        "--calibration_method",
+        type=str,
+        required=False,
+        choices=["awq", "awq_clip"],
+        help="Path of the tokenizer dir.",
```
Comment (tianleiwu) [inline onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py:1358]: update the help message.

File: onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py [hunk 2]
```diff
@@ -1073,10 +1334,35 @@ def parse_args():
         nargs="+",
         required=False,
         help="Key-value pairs in op_type:axis_to_quantize separated by space."
-        "Specify the axis to quantize for an op. Default {MatMul:0, Gather:1}"
-        "Example: --quant_axes MatMul:0 Gather:1",
+             "Specify the axis to quantize for an op. Default {MatMul:0, Gather:1}"
+             "Example: --quant_axes MatMul:0 Gather:1",
+    )
+    # Additional arguments specific to nvidia_awq
```
Comment (tianleiwu) [inline onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py:1340]: May add a parser group like
```
awq_group = parser.add_mutually_exclusive_group()
awq_group.add_argument(...)
```

File: onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py [hunk 3]
```diff
@@ -1073,10 +1334,35 @@ def parse_args():
...
+        "--tokenizer_dir",
+        type=str,
+        required=False,
+        help="Path of the tokenizer dir.",
+    )
+    parser.add_argument(
+        "--calibration_method",
+        type=str,
+        required=False,
+        choices=["awq", "awq_clip"],
```
Comment (tianleiwu) [inline onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py:1357]: nit: how about the following:
```
default="awq_lite",
choices=["awq_lite", "awq_clip"],
```
Then, there is no need to add extra code to post-process the value later.

File: onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py [hunk 4]
```diff
@@ -225,6 +224,216 @@ def __init__(
...
+                                                 trust_remote_code=True)
+        tokenizer.add_special_tokens({'pad_token': '[PAD]'})
+        tokenizer.pad_token = tokenizer.eos_token
+
+        assert calib_size <= max_calib_rows_to_load, "calib size should be no more than max_calib_rows_to_load"
+
+        if "cnn" in dataset_name:
+            dataset2 = load_dataset("cnn_dailymail", name="3.0.0", split="train").select(range(max_calib_rows_to_load))
+            column = "article"
+        elif "pilevel" in dataset_name:
```
Comment (tianleiwu) [inline onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py:377]: why the name is pilevel? It is pile validation set. Suggest to to use name "pile" or "pileval"

Comment (anujj) [reply onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py]: Done

File: onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py (hunk 3 above)
Comment (anujj) [reply onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py:1357]: intentionally kept is like this

File: onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py (hunk 2 above)
Comment (anujj) [reply onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py]: done

File: onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py (hunk 1 above)
Comment (anujj) [reply onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py]: done

**Comment by anujj** [comment]
@tianleiwu : I have addressed the review comments, can we please merge this PR today.

**Review by tianleiwu — DISMISSED** [review]

**Comment by tianleiwu** [comment]
@anujj, please fix format according to https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md#linting

**Comment by anujj** [comment]
Addressed linting issues

**Review by tianleiwu — APPROVED** [review]

**Comment by tianleiwu** [comment]
Thanks for the contribution.
